pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -56,8 +56,12 @@ func (lp *LifecycleProcessor) OnInvokeEnd(endDetails *InvocationEndDetails) {
 
 		if lp.InferredSpansEnabled {
 			log.Debug("[lifecycle] Attempting to complete the inferred span")
-			log.Debugf("[lifecycle] The inferred span attributes are: %s", inferredSpan)
-			inferredSpan.CompleteInferredSpan(lp.ProcessTrace, endDetails.EndTime, endDetails.IsError, TraceID(), SamplingPriority())
+			if inferredSpan.Span.Start != 0 {
```

Comment:
what if inferredSpan.Span is nil?

Reply:
I don't think this is possible:
1. The constants[ are int64](https://github.com/DataDog/datadog-agent/blob/main/pkg/serverless/trace/inferredspan/constants.go#L60-L64), so it will be 0 if the parsing fails 
2.[ span.Start](https://github.com/DataDog/datadog-agent/blob/96fb50274f3a3458ccdafbec3e151a984b67ab1a/pkg/trace/pb/span.pb.go#L50) is int64 so it will also be 0 if unset.

Reply:
Ahhh I misread. Good idea to check that.

Reply:
That's also impossible because we generate the [span](https://github.com/DataDog/datadog-agent/blob/main/pkg/serverless/invocationlifecycle/lifecycle.go#L38) when we call `OnInvokeStart`. That calls `GenerateSpan` [here](https://github.com/DataDog/datadog-agent/blob/ff227400078e7ee5c9c13c10906e9cb7cb3e8b9c/pkg/serverless/trace/inferredspan/inferred_span.go#L154).